extensions/common/context.cc

```diff
@@ -56,8 +56,13 @@ google::protobuf::util::Status extractNodeMetadata(
   }
   google::protobuf::util::JsonParseOptions json_parse_options;
   json_parse_options.ignore_unknown_fields = true;
-  return JsonStringToMessage(metadata_json_struct, node_info,
-                             json_parse_options);
+  status =
+      JsonStringToMessage(metadata_json_struct, node_info, json_parse_options);
+
+  // used to extract a node key at creation time.
+  absl::StrAppend(node_info->mutable_node_key(), node_info->name(), ".",
```

Comment:
Instead of constructing a new key, how about using the node metadata id? It is also available in filter state: https://github.com/istio/proxy/blob/d5701c068d4e04971ba4c640ae27db70b730b632/extensions/stats/plugin.cc#L161

Then in addEdge call, we could add a new param `metadata_id`, which could be used as cache key here.

Reply:
Done.